src/main/java/wolox/training/controllers/UserController.java

```diff
@@ -0,0 +1,148 @@
+	 * @param user: Data structure with the updated fields to update the {@link User}
+	 * @return The updated {@link User}
+	 * @throws UserNotFoundException: When the user id that was passed doesn't belong to any {@link User}
+	 * @throws IdMismatchException: When the id that was passed doesn't match the id that's in within the JSON
+	 * @throws DatabaseException: When the save method throws an error coming from the database
+	 */
+	@PutMapping("{id}")
+	public ResponseEntity<User> update(@PathVariable int id,@Valid @RequestBody User user)
+			throws UserNotFoundException, IdMismatchException, DatabaseException {
+		if(user.getId() != id) {
+			throw new IdMismatchException(ExceptionsConstants.ID_MISMATCH);
+		}
+
+		if(!userRepository.existsById(id)) {
+			throw new UserNotFoundException(ExceptionsConstants.USER_NOT_FOUND);
+		}
+
+		try {
+			return new ResponseEntity<>(userRepository.save(user), HttpStatus.OK);
+		}  catch (Exception e) {
+			throw new DatabaseException(ExceptionsConstants.DATA_SAVE_INTEGRITY_VIOLATION);
+		}
+	}
+
+	/**
+	 * Deletes a specified {@link User} by its id
+	 * @param id: Id of the user
+	 * @return An empty, No Content response
+	 * @throws UserNotFoundException: UserNotFoundException: When the user id that was passed doesn't belong to any {@link User}
+	 */
+	@DeleteMapping("{id}")
+	public ResponseEntity<Void> delete(@PathVariable int id) throws UserNotFoundException {
+		if(!userRepository.existsById(id)) {
+			throw new UserNotFoundException(ExceptionsConstants.USER_NOT_FOUND);
+		}
+		userRepository.deleteById(id);
+		return new ResponseEntity<>(HttpStatus.NO_CONTENT);
+	}
+
+	/**
+	 * Adds or removes a {@link Book} from the user's book list
+	 * @param userId: Id of the {@link User}
+	 * @param bookId: Id of the {@link Book}
+	 * @param action: Action to be performed
+	 * @return The updated {@link User}
+	 * @throws UserNotFoundException: When the user id that was passed doesn't belong to any {@link User}
+	 * @throws BookNotFoundException: When the book id that was passed doesn't belong to any {@link Book}
+	 * @throws BookAlreadyOwnedException: When the specified {@link User} already owns the provided {@link Book}
+	 * @throws ActionNotFoundException: When the provided action doesn't exist
+	 * @throws DatabaseException: When the save method throws an error coming from the database
+	 */
+	@PutMapping("{userId}/books/{bookId}")
+	public ResponseEntity<User> modifyBookList(@PathVariable int userId, @PathVariable int bookId, @RequestParam String action)
+			throws UserNotFoundException, BookNotFoundException, BookAlreadyOwnedException, ActionNotFoundException, DatabaseException {
+		User user = userRepository.findById(userId).orElseThrow(() -> new UserNotFoundException(
+				ExceptionsConstants.USER_NOT_FOUND));
+		Book book = bookRepository.findById(bookId).orElseThrow(() -> new BookNotFoundException(
+				ExceptionsConstants.BOOK_NOT_FOUND));
+
+		if(action.equals("add")) {
```

Comment:
maybe you can add the actions to an Enum

Reply:
Done!